File: scripts/src/entities/definition.ts
import { BaseEntity, Column, Entity, JoinColumn, ManyToOne, PrimaryGeneratedColumn } from "typeorm";
import { Vocab } from "./vocab";

@Entity("definitions")
export class Definition extends BaseEntity {
    @PrimaryGeneratedColumn()
    id: number;

    @Column({ type: "int" })
    vocab_id: number;

    @Column("text", { array: true })
    parts_of_speech: string[];

    @Column("text", { array: true })
    meanings: string[];

    @Column("text", { array: true })
    tags: string[];

    @JoinColumn({ name: "vocab_id" })
    @ManyToOne(() => Vocab, (vocab) => vocab.definitions)
    vocab: Vocab;
}

File: goibackend/src/main/java/com/sebastianaldi17/goibackend/models/Definition.java
package com.sebastianaldi17.goibackend.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;

@Entity(name = "definitions")
@Data
public class Definition {
    @Id
    @GeneratedValue
    @JsonIgnore
    private Long id;

    private String[] parts_of_speech;
    private String[] meanings;
    private String[] tags;

    @ManyToOne
    @JoinColumn(name = "vocab_id")
    @JsonBackReference
    private Vocab vocab;
}
